go/private/actions/link.bzl

```diff
@@ -74,7 +74,9 @@ def emit_link(
         tool_args.add("-race")
     if go.mode.msan:
         tool_args.add("-msan")
-    if (go.mode.static and not go.mode.pure) or go.mode.link != LINKMODE_NORMAL:
+    if ((go.mode.static and not go.mode.pure) or
```

Comment:
Is this change permanent? Could you include a link to the bug where you did the investigation?

Reply:
Yes, this is probably permanent. [#2614 comment](https://github.com/bazelbuild/rules_go/issues/2614#issuecomment-680997524) has more information and links to the upstream issue and CL.

Does this break something for you? Without this fix, race builds that are internally linked will fail on Windows with recent mingw versions. Either the Go internal linker or mingw need to be fixed, but I don't know enough about either to recommend a proper solution.

Reply:
Need to submit this because CI has been red for a few days (since they upgraded mingw), and this is blocking the release. Please let me know soon if this causes problems though.

Reply:
Sorry, I was merely trying to indicate that this is an instance of "magic numbers", or "black magic", or whatever your team calls it. A future reader of this will come onto it and say

"How did this magic combination of booleans come into place that force external... I guess I shouldn't touch it because it's fragile"

I was asking for a comment that said something like:

```
# These magic values come from the fact that windows incorrectly links
# except when using external due to
# https://github.com/bazelbuild/rules_go/issues/2614#issuecomment-680997524
```

Reply:
A comment would be good for each of the conditions. I'll add that tomorrow.